reflection.md

```diff
@@ -2,17 +2,68 @@
 
 ## Which operators were easiest to implement and why?
 
+`||` and `~` were easy because I was just renaming `Union` and `Concat`. We 
+weren't making the actual implementation from scratch or anything.
+
 ## Which operators were most difficult to implement and why?
 
+Making the implicit `string2regex` was challenging because I had to be conscious 
+of the types I was dealing with. I knew the overall strategy, but I tried too 
+hard to make it a one-liner. However, putting all the functionality on one line 
+made it difficult to debug, especially since types didn't implicitly convert 
```

Comment:
It seems reasonable to not use implicit conversion within your implicit conversion function, although it would be kind of cool to do so.